jme3-effects/src/main/java/com/jme3/post/filters/ContrastAdjustmentFilter.java

```diff
@@ -0,0 +1,402 @@
+import com.jme3.export.JmeImporter;
+import com.jme3.export.OutputCapsule;
+import com.jme3.material.Material;
+import com.jme3.post.Filter;
+import com.jme3.renderer.RenderManager;
+import com.jme3.renderer.ViewPort;
+import java.io.IOException;
+
+/**
+ * A filter to adjust the colors of a rendered scene by normalizing each color channel to a specified range,
+ * applying a power law, and scaling the output. The alpha channel is unaffected.
+ *
+ * @author pavl_g.
+ */
+public class ContrastAdjustmentFilter extends Filter {
+
+    protected float redChannelExponent = 1f;
+    protected float greenChannelExponent = 1f;
+    protected float blueChannelExponent = 1f;
+    /**
+     * Lower limit of the input range for all color channels: a level that the filter normalizes to 0.
+     */
+    protected float lowerLimit = 0f;
+    /**
+     * Upper limit of the input range for all color channels: the level that the filter normalizes to 1
+     * (before output scaling).
+     */
+    protected float upperLimit = 1f;
+    //the final pass scale factor
+    protected float redChannelScale = 1f;
+    protected float greenChannelScale = 1f;
+    protected float blueChannelScale = 1f;
+
+    /**
+     * Instantiates a default color contrast filter, default input range and default scale.
+     * Default values :
+     * - Exponents = 1.0f on all channels.
+     * - Input Range Lower Limit = 0f.
+     * - Input Range Upper Limit = 1f.
+     * - Scale = 1.0f on all channels.
+     */
+    public ContrastAdjustmentFilter() {
+        super("Contrast Adjustment");
+    }
+
+    /**
+     * Instantiates a color contrast filter with a specific exponent, default scale and default input range.
+     *
+     * @param exponent an exponent to apply on all channels.
+     */
+    public ContrastAdjustmentFilter(float exponent) {
+        this();
+        this.redChannelExponent = exponent;
+        this.greenChannelExponent = exponent;
+        this.blueChannelExponent = exponent;
+    }
+
+
+    /**
+     * Sets the exponents used to adjust the contrast of the color channels.
+     * Default values are 1f.
+     *
+     * @param redChannelExponent   the red channel exponent.
+     * @param greenChannelExponent the green channel exponent.
+     * @param blueChannelExponent  the blue channel exponent.
+     * @return this filter instance for a chain call.
+     */
+    public ContrastAdjustmentFilter setExponents(float redChannelExponent, float greenChannelExponent,
+            float blueChannelExponent) {
+        setRedExponent(redChannelExponent);
+        setGreenExponent(greenChannelExponent);
+        setBlueExponent(blueChannelExponent);
+
+        return this;
+    }
+
+    /**
+     * Sets the power-law exponent for the red channel.
+     *
+     * @param exponent the desired exponent (default=1)
+     * @return this filter instance, for chaining
+     */
+    public ContrastAdjustmentFilter setRedExponent(float exponent) {
```

Comment:
Could you please standardize the getter and setter methods of the variables? for example `getRedChannelExponent` and `setRedExponent` -> could you change to get/set RedExponent, get/set GreenExponent, get/set BlueExponent and so on for the other class attributes?

Reply:
Well done. I can't resolve the conversation because I didn't open the PR and I don't have write access to the repository on which the PR was opened.